caching/base.py

```diff
@@ -86,16 +71,39 @@ def query_key(self):
         master), throwing a Django ValueError in the process. Django prevents
         cross DB model saving among related objects.
         """
-        query_db_string = 'qs:%s::db:%s' % (self.query_string, self.db)
+        query_db_string = 'qs:%s::db:%s' % (self.queryset.query_key(), self.db)
         return make_key(query_db_string, with_locale=False)
 
+    def cache_objects(self, objects, query_key):
+        """Cache query_key => objects, then update the flush lists."""
+        log.debug('query_key: %s' % query_key)
+        query_flush = flush_key(self.queryset.query_key())
+        log.debug('query_flush: %s' % query_flush)
+        cache.add(query_key, objects, timeout=self.timeout)
+        invalidator.cache_objects(self.queryset.model, objects, query_key, query_flush)
+
     def __iter__(self):
+        if hasattr(super(CacheInternalCommonMixin, self), '__iter__'):
```

Comment:
Perhaps `try:`/`except AttributeError:` to avoid `hasattr()` usage? The Django 1.8 bit can go away eventually, and there will be no performance hit for Django 1.9+

Reply:
+1